Compiler error using CSVWriter

When the CSVWriter is used in a model or if the example of CSVWriter is simulated, I receive the following compiler message:

Compiling and linking the model (Visual C++). 

dsmodel.c
c:\users\u0132350\documents\git\taco\library\ideas-master\ideas\resources\c-sources\fileWriterStructure.c(8) : warning C4018: '<' : signed/unsigned mismatch
C:/Users/u0132350/Documents/GIT/TACO/Library/IDEAS-master/IDEAS/Resources/C-Sources\fileWriterFree.c(55) : warning C4013: 'asprintf' undefined; assuming extern returning int
Creating library dymosim.lib and object dymosim.exp
dsmodel.obj : error LNK2019: unresolved external symbol _asprintf referenced in function _fileWriterFree
dymosim.ex_ : fatal error LNK1120: 1 unresolved externals

Error generating Dymosim.


The function _asprintf is not supported by Windows (https://stackoverflow.com/questions/40159892/using-asprintf-on-windows). Solving this problem could probably solve the issue.

--- a/IBPSA/Resources/C-Sources/fileWriterFree.c
+++ b/IBPSA/Resources/C-Sources/fileWriterFree.c
@@ -51,11 +51,10 @@ void fileWriterFree(void* ptrFileWriter){
   /* If this FileWriter writes in a CombiTimeTable format, prepend the required header
   now that we know how many lines have been written. */
   if (ID->isCombiTimeTable){
-    char* buf;
-    asprintf(&buf,"#1\ndouble csv(%i,%i)\n",ID->numRows,ID->numColumns);
+    char buf[255];
+    sprintf(buf,"#1\ndouble csv(%i,%i)\n",ID->numRows,ID->numColumns);
     prependString(ID->fileWriterName, buf);
-    free(buf);
-    }
+  }
 
   freeBase(ptrFileWriter);
 

